### DISCUSSION TITLE
how can I know the terminal width?

### DISCUSSION THREAD

**Discussion by elfrucool** [pterm/pterm#642] [Q&A] [answered]
when prompting (interactive text input), if the prompt text is bigger than the terminal width, every typed character will duplicate the entire line (this could be considered a bug but that's another story)

to solve this I can put part of the prompt text in a previous printed text, then the actual text input; but if the terminal has enough space, I wish to put all the text on a single line.

To do that, I wish to know the terminal width, to apply either approach (think on what is made in CSS to better format HTML elements on a desktop device and mobile)

**Comment by elfrucool** [comment, marked as answer]
Nevermind, I found it: `pterm.GetTerminalWidth()`

**Reply by MarvinJWendt** [reply]
Hi, we are currently tracking that issue in #560 :)
